cpp/calculate-demand.cpp

```diff
@@ -162,17 +183,17 @@ long calculateUnsatisfiedDemand(std::string targetFilename, std::string demoFile
 #endif
 
     // Second pass: we apply the multiplying factor to each pixel
-    for (int iXBlock = 0; iXBlock < nXBlocks; ++iXBlock) {
+    for (int iXBlock = 0; iXBlock < facilityNXBlocks; ++iXBlock) {
       xOffset = iXBlock*xBlockSize;
       nXValid = xBlockSize;
-      if (iXBlock == nXBlocks-1) nXValid = xSize - xOffset;
+      if (iXBlock == facilityNXBlocks-1) nXValid = facilityXSize - xOffset;
```

Comment:
Can we add an assertion here to double-check that we don't need to check the previous condition (ie `xSize - xOffset` when on the last x block of the region raster)? Since blocks are aligned and the facility's raster should be contained within the region's one, it should never happen, but better be safe than sorry.